front/src/applications/stdcm/utils/fomatConflicts.ts

```diff
@@ -0,0 +1,82 @@
+import dayjs from 'dayjs';
+
+import type { ManageTrainSchedulePathProperties } from 'applications/operationalStudies/types';
+import type { Conflict } from 'common/api/osrdEditoastApi';
+import type { SuggestedOP } from 'modules/trainschedule/components/ManageTrainSchedule/types';
+
+interface ConflictWithInterval {
+  trainIds?: number[];
+  startTime: string;
+  endTime: string;
+  endDate: string;
+  startDate: string;
+  conflictType: string;
+  waypointBefore: string | null;
+  waypointAfter: string | null;
+}
+
+// Helper function to find the closest operational point before or after a given position
+const findClosestOperationalPoint = (
+  operationalPoints: SuggestedOP[],
+  position: number,
+  direction: 'before' | 'after'
+) => {
+  if (direction === 'before') {
+    const pointsBefore = operationalPoints?.filter((point) => point?.positionOnPath <= position);
+    return pointsBefore
+      ? pointsBefore?.reduce((prev, curr) =>
+          curr.positionOnPath > prev.positionOnPath ? curr : prev
+        )
+      : null;
+  }
+  const pointsAfter = operationalPoints?.filter((point) => point.positionOnPath >= position);
+  return pointsAfter
+    ? pointsAfter?.reduce((prev, curr) => (curr.positionOnPath < prev.positionOnPath ? curr : prev))
+    : null;
+};
+
+// Function to get the start and end positions of each conflict
+// eslint-disable-next-line import/prefer-default-export
+export const processConflicts = (
+  conflicts: Conflict[],
+  pathProperties: ManageTrainSchedulePathProperties
```

Comment:
When you have made the `pathProperties` type change
```suggestion
  pathProperties: StdcmPathProperties
```